examples/step-79/step-79.cc

```diff
@@ -0,0 +1,2756 @@
+ *
+ * ---------------------------------------------------------------------
+
+ *
+ * Author: Justin O'Connor, Colorado State University, 2021.
+ */
+
+
+// @sect3{Preliminaries}
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/function.h>
+#include <deal.II/base/tensor.h>
+#include <deal.II/base/timer.h>
+#include <deal.II/base/signaling_nan.h>
+
+#include <deal.II/lac/block_vector.h>
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/block_sparse_matrix.h>
+#include <deal.II/lac/linear_operator.h>
+#include <deal.II/lac/packaged_operation.h>
+#include <deal.II/lac/sparse_direct.h>
+#include <deal.II/lac/affine_constraints.h>
+
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_refinement.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_renumbering.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/fe_dgq.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_q.h>
+
+#include <deal.II/numerics/vector_tools.h>
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/data_out.h>
+
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+
+// Above are fairly normal files to include.  These also include the
+// one for the sparse direct class SparseDirectUMFPACK. This is not
+// the most efficient way to solve large linear problems, but it will
+// do for now.
+//
+// As usual, we put everything into a common namespace. We then start
+// by declaring a number of symbolic names for numbers that will be
+// used throughout. Specifically, we have a *lot* of variables in this
+// program (of course the density and the displacement, but also the
+// unfiltered density and quite a number of Lagrange multipliers. It
+// is easy to forget which of these variables is at which position in
+// the solution vector, and trying to use numbers for these vector
+// components is a prescription for bugs. Rather, we define static
+// variables that can be used in all of these places and that have to
+// be initialized only once. In practice, this will lead to some
+// lengthy expressions, but they are more readable and less likely to
+// be wrong.
+//
+// A similar issue arises with the ordering of blocks in the system
+// matrix and in vectors. The matrices have $9\times 9$ blocks, and
+// it's difficult to remember which is which. It is far easier to just
+// use symbolic names for those as well.
+//
+// Finally, while we're at it, we introduce symbolic names also for
+// the boundary indicators we will use, in the same spirit as was done
+// in step-19.
+//
+// In all of these cases, we declare these variables as members in a
+// namespace. In the case of the solution components, the concrete
+// values of these variables depend on the space dimension, so we use
+// [template
+// variables](https://en.cppreference.com/w/cpp/language/variable_template)
```

Comment:
does markdown work?

Reply:
Yes! I just double checked on my locally built documentation and it's looking like it works fine. :)